sql/core.go

```diff
@@ -255,6 +266,19 @@ type RowIterTypeSelector interface {
 	IsNode2() bool
 }
 
+// DisjointedChildrenNode is a Node that contains multiple, disjointed groupings of child nodes. This is a highly
+// specialized node that will not be applicable to the majority, as most nodes will return all children in the Children
+// function.
+type DisjointedChildrenNode interface {
```

Comment:
This smells funny.

What's important about this disjointedness? Why does it matter if the child nodes are siblings v. groups of siblings? How does it impact analysis to view them in one way or another?

When we have done things like this in the past, we've done it through an intermediary node type, which might be preferable here as well. Something like `plan.ChildContainer` or `plan.ChildGrouping`.

Reply:
Seeing which plan nodes implement this, additionally uneasy. This is the kind of abstraction that gets confused with the original and leaks to where it shouldn't be used.

I would almost prefer to have type switch cases for the 3 plan nodes that need this everywhere required.

Or, if you must keep this abstraction, choose a name that makes it obvious it's for stored procedure analysis, and put it in the plan package instead of alongside the core interfaces, like you've done with other procedure-only interfaces.